Rich-Text-Editor-(RTE)-Overview.md

```diff
@@ -1,117 +1,136 @@
+See [`schema-based-html-editor.directive.html`](https://github.com/oppia/oppia/blob/develop/core/templates/components/forms/schema-based-editors/schema-based-html-editor.directive.html) for a more complicated usage example.
 
 The directive exposes the following attributes:
-* `ng-model`: bind to the variable storing the content the RTE is editing
-* `ui-config` (optional): used to configure the RTE. For example, you can set the placeholder text or whether to hide complex extensions.
 
-# Third Party
-The RTE depends on the following third party libraries, all specified in our `oppia/manifest.json`:
-* [CKEditor](https://github.com/ckeditor/ckeditor-dev): CKEditor is a WYSIWYG rich text editor. The RTE is largely just a wrapper around CKEditor.
+* `[value]`: Bind to the variable storing the content the RTE is editing.
+* `(value-change)`: Function call that handles the user's changes to the RTE.
+* `[ui-config]` (optional): Used to configure the RTE. For example, you can set the placeholder text or whether to hide complex extensions.
+
+## Third-Party Libraries
+
+The RTE depends on the following third-party libraries, all specified in our `manifest.json`:
+
+* [CKEditor 4](https://github.com/ckeditor/ckeditor4): CKEditor is a WYSIWYG rich text editor. The RTE is largely just a wrapper around CKEditor. We currently use version 4.
 * [SharedSpace](https://ckeditor.com/cke4/addon/sharedspace): ensures that the toolbar will always remain in one designated place on the page.
 * [Bootstrap](https://ckeditor.com/cke4/addon/bootstrapck): provides the skin of CKEditor's toolbar.
 
-# Upgrading CKEditor
+### Upgrading CKEditor
+
 As described above, the core of our RTE is the 3rd party library CKEditor. If you are upgrading the CKEditor version we are using, it is important to check that the new CKEditor version does not cause any regressions in the RTE. After upgrading the CKEditor version, use the RTE in the exploration editor to perform the following checks in both the Chrome and Firefox browsers:
-* Input some plaintext and save. Ensure that the content doesn't disappear upon save.
+
+* Input some plain text and save. Ensure that the content doesn't disappear upon save.
 * Input a single Rich Text Component (such as the Math component) and then save. Make sure it is saved properly.
 * Try using backspace to delete the Rich Text Component. Make sure it is actually deleted.
-* Try copy and pasting a Rich Text Component. Hit save and make sure that the component is copied properly.
-* Try drag and dropping a Rich Text Component from one place in the content to another, then saving. Make sure the component is moved properly.
+* Try copying-and-pasting a Rich Text Component. Hit save and make sure that the component is copied properly.
+* Try dragging-and-dropping a Rich Text Component from one place in the content to another, then saving. Make sure the component is moved properly.
```

Comment:
Maybe add a step to make sure that editing the component also works after these steps?

Reply:
done